src/zenml/integrations/gcp/model_deployers/vertex_model_deployer.py

```diff
@@ -0,0 +1,270 @@
+    DEFAULT_DEPLOYMENT_START_STOP_TIMEOUT,
+    BaseModelDeployerFlavor,
+)
+from zenml.services import BaseService, ServiceConfig
+from zenml.stack.stack import Stack
+from zenml.stack.stack_validator import StackValidator
+
+logger = get_logger(__name__)
+
+
+class VertexModelDeployer(BaseModelDeployer, GoogleCredentialsMixin):
+    """Vertex AI endpoint model deployer."""
+
+    NAME: ClassVar[str] = "Vertex AI"
+    FLAVOR: ClassVar[Type["BaseModelDeployerFlavor"]] = (
+        VertexModelDeployerFlavor
+    )
+
+    @property
+    def config(self) -> VertexModelDeployerConfig:
+        """Returns the `VertexModelDeployerConfig` config.
+
+        Returns:
+            The configuration.
+        """
+        return cast(VertexModelDeployerConfig, self._config)
+
+    def setup_aiplatform(self) -> None:
+        """Setup the Vertex AI platform."""
+        credentials, project_id = self._get_authentication()
+        aiplatform.init(
+            project=project_id,
+            location=self.config.location,
+            credentials=credentials,
+        )
+
+    @property
+    def validator(self) -> Optional[StackValidator]:
+        """Validates that the stack contains a model registry.
+
+        Also validates that the artifact store is not local.
+
+        Returns:
+            A StackValidator instance.
+        """
+
+        def _validate_stack_requirements(stack: "Stack") -> Tuple[bool, str]:
+            """Validates that all the stack components are not local.
+
+            Args:
+                stack: The stack to validate.
+
+            Returns:
+                A tuple of (is_valid, error_message).
+            """
+            # Validate that the container registry is not local.
+            model_registry = stack.model_registry
+            if not model_registry and isinstance(
+                model_registry, VertexAIModelRegistry
+            ):
+                return False, (
+                    "The Vertex AI model deployer requires a Vertex AI model "
+                    "registry to be present in the stack. Please add a Vertex AI "
+                    "model registry to the stack."
+                )
+
+            # Validate that the rest of the components are not local.
+            for stack_comp in stack.components.values():
+                # For Forward compatibility a list of components is returned,
+                # but only the first item is relevant for now
+                # TODO: [server] make sure the ComponentModel actually has
+                #  a local_path property or implement similar check
+                local_path = stack_comp.local_path
+                if not local_path:
+                    continue
+                return False, (
+                    f"The '{stack_comp.name}' {stack_comp.type.value} is a "
+                    f"local stack component. The Vertex AI Pipelines "
+                    f"orchestrator requires that all the components in the "
```

Comment:
not an orchestrator here. Also, I don't think the model deployer should actually care whether the other components are local?